Author: Theral Mackey <theral.mackey@gmail.com>
Description: Base64 encode passwords sent to shell (escapeshellarg is horrible)

Abstract: The php escapeshellarg() function is really bad at doing things
  properly. Its better, especially for passwords that might contain ANyth1nG!
  to base64 encode it and pass that value to a shell command. The shell command
  as set in the gosa Hook can then use $(base64_decode(%password)) to safely
  utilize any password sent in without wonky stuff breaking it. See also
  patches 0006, 0007 and 1004 

================================================================================
--- a/gosa-core/include/functions.inc
+++ b/gosa-core/include/functions.inc
@@ -2929,7 +2929,7 @@
             $attrsEvent[$name] = escapeshellarg($value);
         }
-        $attrsEvent['current_password'] = escapeshellarg($old_password);
+        $attrsEvent['current_password'] = base64_encode($old_password);
-        $attrsEvent['new_password'] = escapeshellarg($password);
+        $attrsEvent['new_password'] = base64_encode($password);
 
         // Call the premodify hook now
         $passwordPlugin = new password($config,$dn);

